src/Controller/PageAdminController.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/sonata project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\HelpersBundle\Controller;
+
+use Sonata\BlockBundle\Block\BlockServiceManager;
+use Sonata\BlockBundle\Model\Block;
+use Sonata\HelpersBundle\Block\BlockFilter\BlockFilter;
+use Sonata\PageBundle\Admin\BlockAdmin;
+use Sonata\PageBundle\Controller\PageAdminController as BasePageAdminController;
+use Sonata\PageBundle\Model\Page;
+use Sonata\PageBundle\Page\TemplateManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+/**
+ * Class PageAdminController.
+ *
+ * @author Vincent Laurier <vincent.laurier@ekino.com>
+ * @author Romain Mouillard <romain.mouillard@ekino.com>
+ * @author Ramesses Bonhof Keny <ramesses-bonhof.keny@ekino.com>
+ * @author Christian Kollross <christian.kollross@ekino.com>
+ */
+class PageAdminController extends BasePageAdminController
+{
+    /**
+     * @var BlockFilter
+     */
+    private $blockFilter;
+
+    /**
+     * @var BlockAdmin
+     */
+    private $blockAdmin;
+
+    /**
+     * @var BlockServiceManager
+     */
+    private $blockServiceManager;
+
+    /**
+     * @var TemplateManagerInterface
+     */
+    private $templateManager;
+
+    /**
+     * PageAdminController constructor.
+     *
+     * @param BlockFilter              $blockFilter
+     * @param BlockAdmin               $blockAdmin
+     * @param BlockServiceManager      $blockServiceManager
+     * @param TemplateManagerInterface $templateManager
+     */
+    public function __construct(
+        BlockFilter $blockFilter,
+        BlockAdmin $blockAdmin,
+        BlockServiceManager $blockServiceManager,
+        TemplateManagerInterface $templateManager
+    ) {
+        $this->blockFilter         = $blockFilter;
+        $this->blockAdmin          = $blockAdmin;
+        $this->blockServiceManager = $blockServiceManager;
+        $this->templateManager     = $templateManager;
+    }
+
+    /**
+     * @param Request|null $request
+     *
+     * @throws AccessDeniedException
+     * @throws NotFoundHttpException
+     *
+     * @return Response
+     */
+    public function composeContainerShowAction(Request $request = null): Response
+    {
+        if (false === $this->blockAdmin->isGranted('LIST')) {
+            throw new AccessDeniedException();
+        }
+
+        $id    = $request->get($this->admin->getIdParameter());
```

Comment:
If data from query parameters for instance, better to use `$request->query->get`

Reply:
`$request->query->get($this->admin->getIdParameter())` gives me `null` instead of the ID...

Reply:
dump the request ;) i'm sure you will find your id ;)

Reply:
I give you an example, where your data from?

Reply:
`$request->attributes->get($this->admin->getIdParameter());` would work...
![image](https://user-images.githubusercontent.com/1257143/58267221-7940fa00-7d83-11e9-9ec3-d980c418f22f.png)